packages/components/icon/src/Icon.tsx

```diff
@@ -32,59 +33,39 @@ export type IconVariant =
   | 'white'
   | 'premium';
 
-const sizes: { [key in IconSize]: { [key in 'height' | 'width']: string } } = {
-  xlarge: {
-    height: '48px',
-    width: '48px',
-  },
-  large: {
-    height: '32px',
-    width: '32px',
-  },
-  medium: {
-    height: '24px',
-    width: '24px',
-  },
-  small: {
-    height: '18px',
-    width: '18px',
-  },
-  tiny: {
-    height: '16px',
-    width: '16px',
-  },
-};
-
-const fills: { [key in IconVariant]: string } = {
-  muted: tokens.gray600,
-  negative: tokens.red600,
-  positive: tokens.green600,
-  primary: tokens.blue600,
-  secondary: tokens.gray900,
-  warning: tokens.colorWarning,
-  white: tokens.colorWhite,
-  premium: tokens.purple500,
-};
-
-export type IconInternalProps = CommonProps & {
-  children?: ReactElement | ReactElement[];
-  /**
-   * Determines the size of the icon
-   */
-  size?: IconSize;
-  /**
-   * Whether or not to trim the icon width, i.e. set `width` to `auto`
-   */
-  trimmed?: boolean;
-  /**
-   * Determines the fill color used
-   */
-  variant?: IconVariant;
-  /**
-   * Custom SVG viewBox attribute to use
-   */
-  viewBox?: SVGAttributes<SVGSVGElement>['viewBox'];
-};
+type NewOrOldIconProps =
+  | {
+      // Render the icon with new sizes and styling. Not for general use.
+      isNewIcon: true;
```

Comment:
let's not use the `new` as a version differentiator, since it's too ambiguous. Let's use something more specific like `V2` for clarity and future-proofing the codebase for subsequent updates.

Reply:
Fair point but as a counter I will say that these icons are not v2 or similar, for now they will live side by side with the others. However, the prop is meant to be completely removed at some point, and it should not be used by the public.

Reply:
I'm very open to suggestions for different names though! 😄